cmd/thanos/query-frontend.go

```diff
@@ -111,51 +91,40 @@ func runQueryFrontend(
 	logger log.Logger,
 	reg *prometheus.Registry,
 	tracer opentracing.Tracer,
-	conf *queryFrontendConfig,
+	cfg *config,
 	comp component.Component,
 ) error {
 
-	if len(conf.downstreamURL) == 0 {
-		return errors.New("downstream URL should be configured")
```

Comment:
This is still needed. If no downstreamURL is set, the query frontend will switch to the queue mode, which is not supported by Thanos.

Reply:
thanks re-added